### PR TITLE
Frontend now routes authentication to /login route

### PR DESCRIPTION
**Pull request by klanmiko**

### CODE REVIEW

File: applications-frontend/vue.config.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+    devServer: {
+        proxy: {
+            "/api": {
```
Comment (thomasmunduchira) [inline applications-frontend/vue.config.js]: The proxying is done internally using https://github.com/chimurai/http-proxy-middleware, so using "*" might allow you to consolidate these rules.

File: src/controller/review.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 login_manager = Shared.login_manager
 
 
-@review.route('/review', methods=['GET'])
+@review.route('/api/review', methods=['GET'])
```
Comment (thomasmunduchira) [inline src/controller/review.py]: Backend is still expecting calls at `/review`. Why the change? Same for all below

File: src/controller/auth.py [hunk 3]
```diff
@@ -88,6 +93,13 @@ def google_error(google, error, error_description=None, error_uri=None):
     )
     abort(503, message)
 
+@auth.route('/login', methods=['GET'])
+def login():
+    redirect_url = request.args.get('next')
```
Comment (thomasmunduchira) [inline src/controller/auth.py]: The 'next' argument will not be passed in as a query parameter but instead already set in the session as `next` as configured currently. All you need here is to redirect and then read from `next` in the callback. If `next` is overriden by the time the callback is executed, then we should copy over the session data from `next` into `redirect_url` in `/login` and then read from there in the callback.

File: src/controller/auth.py [hunk 4]
```diff
@@ -74,6 +74,11 @@ def google_logged_in(google, token):
     # login the user account
     login_user(user, remember=True)
 
+    redirect_url = session.get('redirect_url')
+
+    if redirect_url is not None:
+        return redirect(redirect_url)
```
[the remark this answers is not in the document]
Comment (klanmiko) [reply src/controller/auth.py]: https://flask-dance.readthedocs.io/en/latest/signals.html?highlight=oauth_authorized

you lied

File: src/controller/review.py (hunk 2 above)
Comment (klanmiko) [reply src/controller/review.py]: routing /review to the frontend and the backend simultaneously is impossible

File: src/controller/auth.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (klanmiko) [reply src/controller/auth.py]: https://github.com/singingwolfboy/flask-dance/blob/master/flask_dance/consumer/oauth2.py

more proof that redirects will short circuit the thing

File: applications-frontend/vue.config.js (hunk 1 above)
Comment (klanmiko) [reply applications-frontend/vue.config.js]: can't, I don't want the frontend to pass /review into the backend. This happens on login redirect, or any sort of redirect.

Leaving /review out of this proxy list will correctly handle it in the devServer and do vue-router stuff

File: src/controller/auth.py (hunk 3 above)
Comment (klanmiko) [reply src/controller/auth.py]: fixed

**Comment by klanmiko** [comment]
@thomasmunduchira session next doesn't work for frontend. Frontend is purely at / route

File: src/controller/review.py (hunk 2 above)
Comment (thomasmunduchira) [reply src/controller/review.py]: Wait my bad, this is the backend.

File: applications-frontend/vue.config.js (hunk 1 above)
Comment (thomasmunduchira) [reply applications-frontend/vue.config.js]: Got it
